src/layer.rs

```diff
@@ -2088,7 +2094,98 @@ mod tests {
         assert_eq!(child1.parent_span_id, root.span_context.span_id());
         assert_eq!(child2.parent_span_id, child1.span_context.span_id());
 
-        // This is surprising, the parent should be `child1`, but is 'root'.
+        // The parent should be `child1`
         assert_eq!(child3.parent_span_id, child1.span_context.span_id());
     }
+
+    #[test]
+    fn follows_from_adds_link() {
+        use crate::OpenTelemetrySpanExt;
+        let mut tracer = TestTracer::default();
+        let subscriber = tracing_subscriber::registry().with(layer().with_tracer(tracer.clone()));
+
+        let span1_id = tracing::subscriber::with_default(subscriber, || {
+            let span2 = tracing::debug_span!("span2");
+            let span1 = tracing::debug_span!("span1");
+
+            // Ensure that span2 is started
+            let _ = span2.context();
+
+            // Establish follows_from relationship
+            span2.follows_from(&span1);
+
+            // Enter span2 to ensure it's exported
+            let _guard = span2.enter();
+
+            // Get span ID for later verification
+            span1.context().span().span_context().span_id()
+        });
+
+        // Verify that span2 has a link to span1
```

Comment:
Very nitty nit: The verification is happening down on 2133, this is just collecting the data. I had to read through with_data_for a couple times to convince myself this wouldn't throw if the data was missing and don't think this is immediately obvious

Reply:
```suggestion
        // Fetch span2 links
```

Reply:
Thanks, I'll fix that with the other fixes.